src/node.js

```diff
@@ -388,6 +396,53 @@
       nextTickQueue.push(obj);
       tickInfo[kLength]++;
     }
+
+    // Internal callbacks that must be run when there is no more micro-tasks
+    // scheduled for this turn.
```

Comment:
`_lateTick` is too generic. Since we only use it for one thing, we could just take whatever closure captures (promise and reason) and add to a qeueue. The function body can be moved then to `drainLateTickQueue`